homeassistant/components/metoffice/sensor.py

```diff
@@ -108,7 +109,8 @@ class MetOfficeCurrentSensor(Entity):
     def __init__(self, entry_data, hass_data, use_3hourly, sensor_type):
         """Initialize the sensor."""
         self._data = hass_data[METOFFICE_DATA]
-        self._coordinator = hass_data[METOFFICE_COORDINATOR]
+        self._hourly_coordinator = hass_data[METOFFICE_HOURLY_COORDINATOR]
+        self._daily_coordinator = hass_data[METOFFICE_DAILY_COORDINATOR]
```

Comment:
Pretty sure you only need one coordinator here. You can use `use_3hourly` to pick the correct one. The same applies to `weather.py`

Reply:
Done